panel/io/handlers.py

```diff
@@ -151,7 +151,13 @@ def capture_code_cell(cell):
 
     # Remove code comments
     if '#' in cell_out:
-        cell_out = cell_out[:cell_out.index('#')].rstrip()
+        try:
+            # To not remove "#000000"
+            cell_tmp = cell_out[:cell_out.index('#')].rstrip()
+            ast.parse(cell_tmp)
+            cell_out = cell_tmp
+        except SyntaxError:
+            pass
```

Comment:
May make another pass at this with a regex.

Reply:
My first instinct was to remove the if-statement. Then I thought about doing something more advanced like regex, but I felt that would take too much time to do and could lead to more problems, such as multiline strings containing #.

Reply:
Yup, it's not at all straightforward. Worth opening an issue maybe.